connection_test.go

```diff
@@ -471,6 +471,155 @@ func TestClient_Send(t *testing.T) {
 
 		require.Equal(t, closer.Used, true, "client didn't use custom connection")
 	})
+
+	t.Run("ClosedHandler is called when connection is closed", func(t *testing.T) {
+		server, err := NewTestServer()
+		require.NoError(t, err)
+		defer server.Close()
+
+		c, err := connection.New(server.Addr, testSpec, readMessageLength, writeMessageLength, connection.SendTimeout(500*time.Millisecond))
+		require.NoError(t, err)
+
+		// to avoid data race with `isClosedHandlerCalled` when
+		// handling conn closed and checking it in the
+		// require.Eventually
+		var m sync.Mutex
+		var isClosedHandlerCalled bool
+		closedHandler := func(c *connection.Connection) {
+			m.Lock()
+			isClosedHandlerCalled = true
```

Comment:
Do we want to have `+= 1` here to check how many times `closedHandler` is called?

Reply:
I think it's a good idea! While I believe it will be called once, it's good to have a test for it.